EKF/AlphaFilter.hpp

```diff
@@ -32,32 +32,60 @@
  ****************************************************************************/
 
 /**
- * First order "alpha" IIR digital filter
+ * @file AlphaFilter.hpp
+ *
+ * @brief First order "alpha" IIR digital filter also known as leaky integrator or forgetting average.
+ *
  * @author Mathieu Bresciani <brescianimathieu@gmail.com>
+ * @author Matthias Grob <maetugr@gmail.com>
  */
 
 #pragma once
 
 template <typename T>
-class AlphaFilter final {
+class AlphaFilter {
 public:
 	AlphaFilter() = default;
 	~AlphaFilter() = default;
 
-	void reset(const T &val) { _x = val; }
-
-	void update(const T &input, float tau, float dt) {
-		const float alpha = dt / tau;
-		update(input, alpha);
+	/**
+	 * Set filter parameters for time abstraction
+	 *
+	 * Both parameters have to be provided in the same units.
+	 *
+	 * @param sample_interval interval between two samples
+	 * @param time_constant filter time constant determining convergence
+	 */
+	void setParameters(float sample_interval, float time_constant) {
+		setAlpha(sample_interval / (time_constant + sample_interval));
 	}
 
-	void update(const T &input, float alpha) { _x = (1.f - alpha) * _x + alpha * input; }
+	/**
+	 * Set filter parameter alpha directly without time abstraction
+	 *
+	 * @param alpha [0,1] filter weight for the previous state. High value - long time constant.
+	 */
+	void setAlpha(float alpha) { _alpha = alpha; }
+
+	/**
+	 * Set filter state to an initial value
+	 *
+	 * @param sample new initial value
+	 */
+	void reset(const T &sample) { _filter_state = sample; }
+
+	/**
+	 * Add a new raw value to the filter
+	 *
+	 * @return retrieve the filtered result
+	 */
+	virtual void update(const T &sample) { _filter_state = updateCalculation(sample); }
```

Comment:
Why is it virtual? Does it get extended?

Reply:
@MaEtUgR Is the plan to implement the same filter but with a saturation at the input maybe?

Reply:
No, it doesn't get extended yet. I can remove this 👍